de.dlr.sc.virsat.server.test/src/de/dlr/sc/virsat/server/resources/WorkspaceAccessResourceTest.java

```diff
@@ -0,0 +1,131 @@
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.UriBuilder;
+
+import org.apache.commons.io.FileUtils;
+import org.eclipse.core.runtime.Status;
+import org.glassfish.jersey.client.ClientConfig;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import de.dlr.sc.virsat.server.Activator;
+import de.dlr.sc.virsat.server.data.VirSatGitAccess;
+import de.dlr.sc.virsat.server.test.AGitAndJettyServerTest;
+
+public class WorkspaceAccessResourceTest extends AGitAndJettyServerTest {
+
+	private File pathToTempLocalRepository1;
+	private File pathToTempLocalRepository2;
+
+	@Before
+	public void setUp() throws Exception {
+		super.setUp();
+		
+		pathToTempLocalRepository1 = File.createTempFile("VirSatLocalRepo1_", "");
+		pathToTempLocalRepository2 = File.createTempFile("VirSatLocalRepo2_", "");
+		Files.delete(pathToTempLocalRepository1.toPath());
+		Files.delete(pathToTempLocalRepository2.toPath());
+		Files.createDirectory(pathToTempLocalRepository1.toPath());
+		Files.createDirectory(pathToTempLocalRepository2.toPath());
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		try {
+			FileUtils.deleteDirectory(pathToTempLocalRepository1);
+			FileUtils.deleteDirectory(pathToTempLocalRepository2);
+		} catch (Exception e) {
+			Activator.getDefault().getLog().log(new Status(Status.INFO, Activator.getPluginId(), "Failed to remove temp directory in test" + e.getMessage()));
+		}
+		super.tearDown();
+	}
+
+	@Test
+	public void testCloneCommitAndUpdateRepository() throws IOException {
+		ClientConfig config = new ClientConfig();
+		Client client = ClientBuilder.newClient(config);
+		
+		URI uri = UriBuilder.fromUri("http://localhost:8000/").build();
+		WebTarget target = client.target(uri);
+		
+		// Clone to local repository 1
+		String result = target.
+			path("/rest").
+			path(WorkspaceAccessResource.PATH_PERSISTENCE).
+			path(WorkspaceAccessResource.PATH_CLONE).
+			queryParam(WorkspaceAccessResource.PARAM_REMOTE, pathToTempUpstreamRepository).
+			queryParam(WorkspaceAccessResource.PARAM_LOCAL, pathToTempLocalRepository1).
+			request().
+			get(String.class);
+		
+		assertEquals("Cloning local repository 1 succeeded", pathToTempLocalRepository1.toString(), result);
+		
+		// Clone to local repository 2
+		String result2 = target.
+			path("/rest").
+			path(WorkspaceAccessResource.PATH_PERSISTENCE).
+			path(WorkspaceAccessResource.PATH_CLONE).
+			queryParam(WorkspaceAccessResource.PARAM_REMOTE, pathToTempUpstreamRepository).
+			queryParam(WorkspaceAccessResource.PARAM_LOCAL, pathToTempLocalRepository2).
+			request().
+			get(String.class);
+		
+		assertEquals("Cloning local repository 2 succeeded", pathToTempLocalRepository2.toString(), result2);
+				
+		// Add a file to repo one and commit it
+		// now add a file to the local repository
+		File newFile = new File(pathToTempLocalRepository1.getAbsolutePath().toString() + "/test.dat");
+		Files.createFile(newFile.toPath());

```

Comment:
Is there any reason why newFile.createNewFile(); doesnt work?
I think it would be a little bit cleaner.

Reply:
Will give it a try